symengine/pow.cpp

```diff
@@ -90,11 +87,22 @@ int Pow::compare(const Basic &o) const
 
 RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
 {
+    if (eq(*a, *zero)) {
+        if (is_a_Number(*b)) {
+            if(static_cast<const Number &>(*b).is_negative())
+                throw std::runtime_error("pow: Zero cannot be raised to negative power");
+            else if(static_cast<const Number &>(*b).is_zero())
+                throw std::runtime_error("Indeterminate");
+            else
+                return zero;
+        }
+        else
+            return zero;
+    }
```

Comment:
This shouldn't be at the top. This should be after the check `eq(*b, *one)` or just before that